source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+         But that function uses getJSONSchema and passes it a dataKey id in order to create a client which uses the JSONSchema
+         we want to use. 
+
+         .. code-block:: java
+
+            import com.mongodb.*;
+            import com.mongodb.client.MongoClient;
+            import com.mongodb.client.MongoClients;
+            import com.mongodb.client.MongoCollection;
+            import com.mongodb.client.model.vault.DataKeyOptions;
+            import com.mongodb.client.vault.ClientEncryptions;
+
+            import org.bson.BsonDocument;
+
+            import java.util.Base64;
+            import java.util.Map;
+
+            import org.json.*;
+
+            public class FieldLevelEncryptionDemo {
+                static String dbName = "fieldLevelEncryptionDemo001";
+                static String collectionName = "patients";
+                public static void main(String[] args) {
+                    MongoCollection secureCollection = createSecureClient().getDatabase(dbName).getCollection(collectionName);
+                    ...
+                }
+                public static JSONObject getEncryptedField(String keyId, String chosenBsonType, Boolean isDeterministic){
+                    JSONObject binaryForKey = new JSONObject();
+                    binaryForKey.put("base64", keyId);
+                    binaryForKey.put("subType", "04");
+                    JSONObject binWrapper = new JSONObject();
+                    binWrapper.put("$binary", binaryForKey);
+                    
+                    JSONObject encryptedFieldWrap = new JSONObject();
+                    JSONObject encrypt = new JSONObject();
+
+                    encrypt.put("bsonType", chosenBsonType);
+                    if(isDeterministic){
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic");
+                    }else{
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Random");
+                    }
+                    encrypt.put("keyId", new JSONArray().put(binWrapper));
+                    encryptedFieldWrap.put("encrypt", encrypt);
+                    return encryptedFieldWrap;
+                }
+                public static JSONObject getStandardField(String chosenBsonType){
+                    JSONObject standardField = new JSONObject();
+                    standardField.put("bsonType", chosenBsonType);
+                    standardField.put("description", "must be a " + chosenBsonType + " and is required");
+                    return standardField;
+                }
+                public static String getJSONSchema(String keyId){
+                    JSONObject properties = new JSONObject();
+                    properties.put("fullName", getStandardField("string"));
+                    properties.put("address", getEncryptedField(keyId, "string", false));
+                    properties.put("ssn", getEncryptedField(keyId, "int", true));
```

Comment:
If we decide on `ssn_last4`, could we stick to snake case for the rest of the fields to stay consistent?

Reply:
I suggested snake case here to demonstrate that `ssn_last4` is a modified version of the `ssn` field rather than a distinct, independent field. That said if it isn't clear feel free to disregard.

Reply:
I'm going to executive decision this and say we stick to camel case. I think we could go back and fourth of snake case vs camel case forever and never reach a conclusion. But ultimately, my reasons for sticking to camel case boils down to:
1. Consistency among this page (if we change this to snake case, this entire page will have to be snake case)2
2. Consistency amongst docs-ecosystem in general
3. last4SSN/ssn_last4 isn't really a modified version of ssn (as in, the reader isn't going to interact with ssn to get last4SSN, they are just two separate fields which are put in a document)